### ISSUE TITLE
[0.28.4+] Moved column displays incorrect values

### ISSUE THREAD

**Issue by budnix** [handsontable/handsontable#4470]
### Description
Moved column displays incorrect values.  

### Steps to reproduce

1. Move last column ("I") to the beginning of the table (before column "A").
2. In the first column, you will see values from the second column (values are doubled).

### Demo
http://jsfiddle.net/9mr7hbpn/

### Your environment
* Handsontable version: 0.34.0
* Browser Name and version: any
* Operating System: any

**Comment by AMBudnik** [comment]
Updated demo (v1.15.0/CE 0.35.0) http://jsfiddle.net/6vuhuufj/
It looks like the `afterColumnMove` hook is returning false data in the first parameter. It returns `null` when it should return the column index.

**Comment by pnowak** [comment]
The error appears because there is one column too much, or on the other hand, the team is missing in the date. If there is the same amount of data and columns error doesn't appear.

http://jsfiddle.net/6vuhuufj/1/
http://jsfiddle.net/6vuhuufj/3/

**Comment by AMBudnik** [comment]
Hi @budnix!
@pnowak is right, there's one extra column key `team` which was not defined anywhere in the source data.
One of our Clients has also reported that he struggles with this issue so if the cause is the same we will be able to close the issue.

**Comment by mountaineerfan5** [comment]
I'm not sure if the incorrect index you were referring to is the columns parameter in the afterColumnMove hook, but the parameter appears to be giving the physical index and not the visual index as it says in the documentation.

http://jsfiddle.net/mountaineerfan5/6vuhuufj/4/

1. Move a column to a new position
2. Move that same column that was just moved to a new position

The columns parameter is the same in both calls to the afterColumnMove hook even though the visual indices are different. So my guess is it is grabbing the physical index. Is it suppose to behave this way?

**Comment by AMBudnik** [comment]
Hi @mountaineerfan5 
it is a bug. The `afterColumnMove` should use visual indexes.

**Comment by AMBudnik** [comment]
The last working version was [`0.27.0`](http://jsfiddle.net/533u62q6/).

![27](https://user-images.githubusercontent.com/10757813/39178912-3cfcc102-47b3-11e8-9a95-1f25629769cf.gif)

Then in [`0.28.0` ](http://jsfiddle.net/L1m607Lv/)the last column's data disappeared(?) but data hasn't been duplicated.

![28](https://user-images.githubusercontent.com/10757813/39179009-7e4d8114-47b3-11e8-83e7-db9a7d9ec423.gif)

From version[ `0.28.4` ](http://jsfiddle.net/f9p1r6v3/)we can observe the same issue as reported in the first comment.

![284](https://user-images.githubusercontent.com/10757813/39179187-fb69a6be-47b3-11e8-9819-a6c7d557a5e2.gif)

**Comment by AMBudnik** [comment]
The issue is still replicable in v 5.0.1

**Comment by AMBudnik** [comment]
Sam example with columns moved programmatically
https://jsfiddle.net/handsoncode/ksfc38t6/12/

for
```js  
const myData = [
  [1, 2, 3],
  [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
]
...
manualColumnMove: [2, 1, 3, 6, 7, 5, 8, 0, 9, 4],
```
we get
![image](https://user-images.githubusercontent.com/10757813/51108297-01216100-17f2-11e9-92d6-c3ec67d82a73.png)

It looks like the plugin is set to understand number of columns defined in the first row.

**Comment by NachoMoment** [comment]
Any ETA on a fix for this?

**Comment by wojciechczerniak** [comment]
No firm date yet, but we're working on it as we speak #5945

**Comment by NachoMoment** [comment]
Awesome, thanks for the update :)

**Comment by aninde** [comment]
This issue looks fixed by the new `index mapper`. Changes will be released soon.
@NachoMoment,  we are currently testing a version 8.0.0.rc with many breaking changes, I recommend observing information from issue [#5751](https://github.com/handsontable/handsontable/issues/5751). 

**Demo**
@mountaineerfan5's version
ver 1.15.1 http://jsfiddle.net/aninde/1g3bp6sa/ 
ver 7.1.1.  http://jsfiddle.net/aninde/8j5auynh/ 
Ver. 8.0.0rc http://jsfiddle.net/aninde/ya0e3wq7/ - fixed

@AMBudnik's version
ver. 6.2.2 https://jsfiddle.net/aninde/cvh32g57/
Ver 7.1.1 https://jsfiddle.net/aninde/2ncsbvr5/
Ver. 8.0.0rc https://jsfiddle.net/aninde/r8zb1stp/ - fixed

**Environment**
Handsontable version:8.0.0.rc
Browser Name and version: Google Chrome version 76.0.3809.100
Operating System: macOS Mojave 10.14.6

**Comment by AMBudnik** [comment]
Great to see this solved in the official 8.0.0-beta1 http://jsfiddle.net/bdw6tagq/

**Comment by aninde** [comment]
This issue still works well on stable, released `v8.0.0` http://jsfiddle.net/aninde/7qjtg54w/
http://jsfiddle.net/aninde/fu0h5okb/.

Closing as solved.

**Comment by AMBudnik** [comment]
Hey @NachoMoment @mountaineerfan5 

as @aninde wrote - this issue should be officially fixed in v 8.0.0 so I hope that you'll be able to update and benefit from this fix.

CDN https://cdn.jsdelivr.net/gh/handsontable/handsontable@latest/dist/handsontable.full.css
https://cdn.jsdelivr.net/gh/handsontable/handsontable@latest/dist/handsontable.full.js
NPM npm i handsontable
Release Notes https://github.com/handsontable/handsontable/releases
